src/lib.rs

```diff
@@ -717,6 +717,325 @@ where
+                Ratio::new(T::from(self) - rhs.numer, rhs.denom)
+            }
+        }
+
+        #[cfg(feature = "bigint")]
+        impl<'a> Sub<&'a BigRational> for $primitive {
+            type Output = BigRational;
+
+            #[inline]
+            fn sub(self, rhs: &'a BigRational) -> BigRational {
+                Ratio::new(BigInt::from(self) - rhs.numer.clone(), rhs.denom.clone())
+            }
+        }
+
+        impl<T> Mul<Ratio<T>> for $primitive
+        where
+            T: Clone + Integer + ::core::convert::From<$primitive>,
+        {
+            type Output = Ratio<T>;
+
+            #[inline]
+            fn mul(self, rhs: Ratio<T>) -> Ratio<T> {
+                Ratio::new(rhs.numer * T::from(self), rhs.denom)
+            }
+        }
+
+        #[cfg(feature = "bigint")]
+        impl<'a> Mul<&'a BigRational> for $primitive {
+            type Output = BigRational;
+
+            #[inline]
+            fn mul(self, rhs: &'a BigRational) -> BigRational {
+                Ratio::new(rhs.numer.clone() * BigInt::from(self), rhs.denom.clone())
+            }
+        }
+
+        impl<T> Div<Ratio<T>> for $primitive
+        where
+            T: Clone + Integer + ::core::convert::From<$primitive>,
+        {
+            type Output = Ratio<T>;
+
+            #[inline]
+            fn div(self, rhs: Ratio<T>) -> Ratio<T> {
+                Ratio::new(rhs.numer, rhs.denom * T::from(self))
+            }
+        }
+
+        #[cfg(feature = "bigint")]
+        impl<'a> Div<&'a BigRational> for $primitive {
+            type Output = BigRational;
+
+            #[inline]
+            fn div(self, rhs: &'a BigRational) -> BigRational {
+                Ratio::new(rhs.numer.clone(), rhs.denom.clone() * BigInt::from(self))
+            }
+        }
+    };
+}
+
+// Implements for isize and BigInt
+impl_primitive_ops_ratio!(Ratio<T> for isize);
+// Implements for BigInt
+impl_primitive_ops_ratio!(Ratio<T> for usize);
+// Implements for BigInt, i64, i32, i8
+impl_primitive_ops_ratio!(Ratio<T> for i8);
+// Implements for isize, i32, i64, BigInt
+impl_primitive_ops_ratio!(Ratio<T> for u8);
+// Implements for isize, i32, i64, BigInt
+impl_primitive_ops_ratio!(Ratio<T> for i16);
+// Implements for i32, i64, BigInt
+impl_primitive_ops_ratio!(Ratio<T> for u16);
+// Implements for i32, i64, BigInt
+impl_primitive_ops_ratio!(Ratio<T> for i32);
+// Implements for i64, BigInt
+impl_primitive_ops_ratio!(Ratio<T> for u32);
+// Implements for i64, BigInt
+impl_primitive_ops_ratio!(Ratio<T> for i64);
+// Implements for BigInt
+impl_primitive_ops_ratio!(Ratio<T> for u64);
+// Implements for BigInt
+#[cfg(has_i128)]
+impl_primitive_ops_ratio!(Ratio<T> for i128);
+// Implements for BigInt
+#[cfg(has_i128)]
+impl_primitive_ops_ratio!(Ratio<T> for u128);
+#[cfg(feature = "bigint")]
+impl_primitive_ops_ratio!(Ratio<T> for BigInt);
+
+#[cfg(feature = "bigint")]
+macro_rules! impl_ops_bigrational_for_float {
```

Comment:
Hmm.  I don't think it's a good idea to make float-conversions implicit, especially since you have to output an `Option`.    That makes it pretty unusual from what most people would expect from ops.  I think it's better to leave this one to the user to explicitly do their float->ratio conversion before attempting ops.